frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.stories.tsx

```diff
@@ -0,0 +1,227 @@
+// @ts-expect-error There is no type definition
+import createAsyncCallback from "@loki/create-async-callback";
+import type { ComponentStory, Story } from "@storybook/react";
+import { useEffect, type ComponentProps } from "react";
+import { Provider } from "react-redux";
+
+import { getStore } from "__support__/entities-store";
+import { getNextId } from "__support__/utils";
+import { NumberColumn, StringColumn } from "__support__/visualizations";
+import { explicitSizeRefreshModeContext } from "metabase/components/ExplicitSize/context";
+import { publicReducers } from "metabase/reducers-public";
+import TABLE_RAW_SERIES from "metabase/visualizations/components/TableSimple/stories-data/table-simple-orders-with-people.json";
+import {
+  createMockCard,
+  createMockColumn,
+  createMockDashboard,
+  createMockDashboardCard,
+  createMockDataset,
+  createMockDatasetData,
+  createMockParameter,
+} from "metabase-types/api/mocks";
+import {
+  createMockDashboardState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { PublicOrEmbeddedDashboardView } from "./PublicOrEmbeddedDashboardView";
+
+export default {
+  title: "embed/PublicOrEmbeddedDashboardView",
+  component: PublicOrEmbeddedDashboardView,
+  decorators: [
+    ReduxDecorator,
+    FasterExplicitSizeUpdateDecorator,
+    WaitForResizeToStopDecorator,
+  ],
+};
+
+function ReduxDecorator(Story: Story) {
+  return (
+    <Provider store={store}>
+      <Story />
+    </Provider>
+  );
+}
+
+function FasterExplicitSizeUpdateDecorator(Story: Story) {
+  return (
+    <explicitSizeRefreshModeContext.Provider value="none">
+      <Story />
+    </explicitSizeRefreshModeContext.Provider>
+  );
+}
+
+/**
+ * This is an arbitrary number, it should be big enough to pass CI tests.
+ * This value works together with FasterExplicitSizeUpdateDecorator which
+ * make sure we finish resizing any ExplicitSize components the fastest.
+ */
+const TIME_UNTIL_ALL_ELEMENTS_STOP_RESIZING = 1000;
```

Comment:
Given you've changed the explicit size refresh mode to none, do we need this timeout? 1s per every component that uses ExplicitSize can be significant

Reply:
It does need this value still. If you run Storybook and visit `frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.stories.tsx` locally. You could still see the resize in action. It's faster but the delay before the layout has stopped shifting is still there.

I tried 1 second before the `ExplicitSize` change, and CI failed. With the `ExplicitSize` change, this passes, so I want to keep it as is.

Here's my test results from running the test 3 times for each scenario locally

M1 Macbook pro 10 cores

#### When setting this timeout to 0 (not removing it for simplicity)
- Took 62s ~21s each
- All 3 iterations failed with either 1-2 different snapshots

#### When setting this timeout to 300 (ExplicitSize timeout)
Same result
- Took 60s ~20s each
- All 3 iterations failed with either 1-2 different snapshots

#### When setting this timeout to 1000 (this PR)

- Took 63s ~ 21s each
- All tests passed


I couldn't find a way to mimic 4 cores on GitHub actions (I tried to run the test inside a docker with `--cpuset-cpus=0-3`, but couldn't figure out how to run Loki inside the container as it wanted to run docker inside docker), so I tested with repeating stories so I have 20 stories for 10 cores and that should be comparable to 8 stories (this PR) for the 4-core GitHub runner

#### When setting this timeout to 0 (not removing it for simplicity)
- Took 88s ~29s each
- All 3 iterations failed with either 1-2 different snapshots

#### When setting this timeout to 300 (ExplicitSize timeout)
Same result
- Took 87s ~29s each
- All 3 iterations failed with either 1-2 different snapshots

#### When setting this timeout to 1000 (this PR)
- Took 90s ~ 30s each
- All tests passed

It's not at all head-to-head comparison, but the big picture I want to convey is having 1 second delay doesn't seem to impact the overall test duration significantly. It's roughly between 1-3% from my test.

Reply:
Got it. Looks good, thanks for benchmarking!